src/Compilers/CSharp/Test/Semantic/Semantics/RecordStructTests.cs

```diff
@@ -5190,5 +5314,1050 @@ public void ToString_UserDefinedPrintMembers_Static()
+            public int FireCount500;
+            public int FireCount1000;
+            public int FireCount4000;
+            public int FireCount5000;
+
+            private static readonly DiagnosticDescriptor Descriptor =
+               new DiagnosticDescriptor("XY0000", "Test", "Test", "Test", DiagnosticSeverity.Warning, true, "Test", "Test");
+
+            public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+            => ImmutableArray.Create(Descriptor);
+
+            public override void Initialize(AnalysisContext context)
+            {
+                context.RegisterCodeBlockStartAction<SyntaxKind>(Handle);
+            }
+
+            private void Handle(CodeBlockStartAnalysisContext<SyntaxKind> context)
+            {
+                switch (context.OwningSymbol.ToTestDisplayString())
+                {
+                    case "A..ctor([System.Int32 X = 0])":
+                        switch (context.CodeBlock)
+                        {
+                            case RecordStructDeclarationSyntax { Identifier: { ValueText: "A" } }:
+                                Interlocked.Increment(ref FireCount100);
+                                break;
+                            default:
+                                Assert.True(false);
+                                break;
+                        }
+                        break;
+                    case "System.Int32 A.M()":
+                        switch (context.CodeBlock)
+                        {
+                            case MethodDeclarationSyntax { Identifier: { ValueText: "M" } }:
+                                Interlocked.Increment(ref FireCount400);
+                                break;
+                            default:
+                                Assert.True(false);
+                                break;
+                        }
+                        break;
+                    case "A..ctor(System.String S)":
+                        switch (context.CodeBlock)
+                        {
+                            case ConstructorDeclarationSyntax { Identifier: { ValueText: "A" } }:
+                                Interlocked.Increment(ref FireCount500);
+                                break;
+                            default:
+                                Assert.True(false);
+                                break;
+                        }
+                        break;
+                    default:
+                        Assert.True(false);
+                        break;
+                }
+
+                context.RegisterSyntaxNodeAction(Handle1, SyntaxKind.NumericLiteralExpression);
+                context.RegisterSyntaxNodeAction(Handle2, SyntaxKind.EqualsValueClause);
+                context.RegisterSyntaxNodeAction(Fail, SyntaxKind.BaseConstructorInitializer);
+                context.RegisterSyntaxNodeAction(Fail, SyntaxKind.ConstructorDeclaration);
```

Comment:
>SyntaxKind.ConstructorDeclaration [](start = 55, length = 33)

Why not test these two?

Reply:
Added. Thanks

---
In reply to: [605218435](https://github.com/dotnet/roslyn/pull/52256#discussion_r605218435) [](ancestors = 605218435)